### ISSUE TITLE
Make it possible to slot native elements instead of just custom ones in cases where the semantics match

### ISSUE THREAD

**Issue by LeaVerou** [shoelace-style/webawesome#307]
_(I'd swear we had an issue for this but I can't find it to save my life)_

I see this as a complement to #217 , if we support styling natives, we should also support slotting them in in the same way as our own components.

This includes things like:
- `<button>` in `<wa-dropdown>`
- `<hr>` in `<wa-menu>`
- `<option>` in `<wa-select>` (this may be too hard to do though, so possibly not worth it)
- ...

**Comment by claviska** [comment]
I agree in principal. Many of these will just work but might need their types updated. For some, like `<hr>`, slotted styles may be required.

As you suspect, `<option>` inside of `<wa-select>` will indeed be a challenge that I'd prefer not to waste time on.
